lua/nvim-treesitter/install.lua

```diff
@@ -302,100 +298,119 @@ local function install_lang(lang, ask_reinstall, cache_folder, install_folder, w
-    end
-    for _, lang in pairs(parsers_to_update) do
-      install({
-        ask_reinstall = 'force',
-        exclude_configured_parsers = true
-      })(lang)
+function M.update(options)
+  options = options or {}
+  return function(bang)
+    return function (...)
+      M.lockfile = {}
+      reset_progress_counter()
+      if ... and ... ~= 'all' then
+        local languages = vim.tbl_flatten({...})
+        local installed = 0
+        for _, lang in ipairs(languages) do
+            if (not is_installed(lang)) or (needs_update(lang)) then
+              installed = installed + 1
+              install({ with_sync = options.with_sync })('!')(lang)
+            end
+        end
+        if installed == 0 then
+          print('Parsers are up-to-date!')
+        end
+      else
+        local parsers_to_update = configs.get_update_strategy() == 'lockfile'
+          and outdated_parsers()
+          or info.installed_parsers()
+        if #parsers_to_update == 0 then
+          print('All parsers are up-to-date!')
+        end
+        for _, lang in pairs(parsers_to_update) do
+          install({
+            exclude_configured_parsers = true,
+            with_sync = options.with_sync
+          })('!')(lang)
+        end
+      end
     end
   end
 end
 
-function M.uninstall(lang)
-  local path_sep = '/'
-  if fn.has('win32') == 1 then
-    path_sep = '\\'
-  end
-
-  if vim.tbl_contains({'all', 'maintained'}, lang) then
-    reset_progress_counter()
-    local installed = info.installed_parsers()
-    if lang == "maintained" then
-      local maintained = parsers.maintained_parsers()
-      installed = vim.tbl_filter(function(l) return vim.tbl_contains(maintained, l) end, installed)
-    end
-    for _, lang in pairs(installed) do
-      M.uninstall(lang)
+function M.uninstall(bang)
+  return function (lang)
+    local path_sep = '/'
+    if fn.has('win32') == 1 then
+      path_sep = '\\'
     end
-  elseif lang then
-    local install_dir, err = utils.get_parser_install_dir()
-    if err then return api.nvim_err_writeln(err) end
 
-    local parser_lib = install_dir..path_sep..lang..".so"
+    if vim.tbl_contains({'all', 'maintained'}, lang) then
+      reset_progress_counter()
+      local installed = info.installed_parsers()
+      if lang == "maintained" then
+        local maintained = parsers.maintained_parsers()
+        installed = vim.tbl_filter(function(l) return vim.tbl_contains(maintained, l) end, installed)
+      end
+      for _, langitem in pairs(installed) do
+        M.uninstall(langitem)
+      end
+    elseif lang then
+      local install_dir, err = utils.get_parser_install_dir()
+      if err then return api.nvim_err_writeln(err) end
 
-    local command_list = {
-      shell.select_rm_file_cmd(parser_lib, "Uninstalling parser for "..lang)
-    }
-    M.iter_cmd(command_list, 1, lang, 'Treesitter parser for '..lang..' has been uninstalled')
+      local parser_lib = install_dir..path_sep..lang..".so"
```

Comment:
Can you use `join_path` from the `utils.lua` module here?

Reply:
possibly? This was taken from moving around existing code so I'm hesitant to rework it under this PR.

Reply:
@smhc can you remember to do this in a follow up PR?

Reply:
Honestly I am unlikely to get to it - it is code I didn't touch, it just got indented. With my follow up changes it should actually show the line as unchanged. I'm not a lua expert to be in a position to clean up based on best practices.